src/main/java/uk/ac/ebi/eva/pipeline/io/readers/VcfReader.java

```diff
@@ -0,0 +1,253 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.io.readers;
+
+import htsjdk.variant.vcf.VCFCompoundHeaderLine;
+import htsjdk.variant.vcf.VCFConstants;
+import htsjdk.variant.vcf.VCFContigHeaderLine;
+import htsjdk.variant.vcf.VCFFileReader;
+import htsjdk.variant.vcf.VCFFilterHeaderLine;
+import htsjdk.variant.vcf.VCFFormatHeaderLine;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import htsjdk.variant.vcf.VCFSimpleHeaderLine;
+import org.opencb.biodata.formats.variant.vcf4.VcfAlternateHeader;
+import org.opencb.biodata.formats.variant.vcf4.VcfFilterHeader;
+import org.opencb.biodata.formats.variant.vcf4.VcfFormatHeader;
+import org.opencb.biodata.formats.variant.vcf4.VcfInfoHeader;
+import org.opencb.biodata.models.variant.Variant;
+import org.opencb.biodata.models.variant.VariantSource;
+import org.springframework.batch.item.file.FlatFileItemReader;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import uk.ac.ebi.eva.pipeline.io.GzipLazyResource;
+import uk.ac.ebi.eva.pipeline.io.mappers.VcfLineMapper;
+import uk.ac.ebi.eva.utils.CompressionHelper;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * VCF file reader.
+ * <p>
+ * This Reader uses a {@link VcfLineMapper} to parse each line, and {@link VCFFileReader} from htslib to fill the
+ * VariantSource with metadata from the VCF header.
+ * <p>
+ * It doesn't matter if the file is compressed or not.
+ */
+public class VcfReader extends FlatFileItemReader<List<Variant>> {
+    /**
+     * You can retrieve the header of the VCF like this: `source.getMetadata().get(VARIANT_FILE_HEADER_KEY)`,
+     * but why "variantFileHeader"? we are using the converter
+     * {@link org.opencb.opencga.storage.mongodb.variant.DBObjectToVariantSourceConverter}, which requires it, and will
+     * store it in mongo as "header".
+     */
+    public static final String VARIANT_FILE_HEADER_KEY = "variantFileHeader";
+
+    private final VariantSource source;
+
+    private final File file;
+
+    public VcfReader(VariantSource source, File file) throws IOException {
+        this.source = source;
+        this.file = file;
+        Resource resource;
+        if (CompressionHelper.isGzip(file)) {
+            resource = new GzipLazyResource(file);
+        } else {
+            resource = new FileSystemResource(file);
+        }
+        setResource(resource);
+        setLineMapper(new VcfLineMapper(source));
+    }
+
+    public VcfReader(VariantSource source, String file) throws IOException {
+        this(source, new File(file));
+    }
+
+    @Override
+    protected void doOpen() throws Exception {
+        prepareVariantSource();
```

Comment:
well, it drops it [here](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/variant/vcf/VCFHeader.java#L115) an hardcodes it [here](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/variant/vcf/VCFHeader.java#L406) which is called if we retrieve the rows in the header.

As alternative, the [toString](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/variant/vcf/VCFHeader.java#L406) method  doesn't put the version, but we would need to reparse it to remove the tabs at the beginning of the line and the extra strings, and prepend the version which should be read by us.

conclusion, if it's a hard restriction, it's a serious argument against using htsjdk.